### PR TITLE
fix #17755 web request

### LINKED ISSUE #17755: the bug of Request class
when
$trustedHosts = [
         '10.0.0.1' => ['X-Forwarded-For']
]

https://github.com/yiisoft/yii2/blob/c02060e2412442513c2a959eb2ebc9ca3ea0c5cd/framework/web/Request.php#L1870
the code will be crash app.

### PR DESCRIPTION
**Pull request by shushenghong**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #17755

### CODE REVIEW

**Comment by samdark** [comment]
@kamarton, @mikk150 would you please take a look?

**Comment by mikk150** [comment]
LGTM

**Review by samdark — APPROVED** [review]

File: tests/framework/web/RequestTest.php [hunk 1]
```diff
@@ -562,9 +581,23 @@ public function testGetIsSecureConnection($server, $expected)
                 'forwarded',
             ],
         ]);
-        $_SERVER = $server;
+        $this->assertEquals($expected, $request->getIsSecureConnection());
 
+        $request = new Request([
+            'trustedHosts' => [
+                '192.168.0.0/24' => ['Front-End-Https', 'X-Forwarded-Proto', 'forwarded'],
```
Comment (kamarton) [inline tests/framework/web/RequestTest.php]: add test with 
```php
'192.168.0.0/24' => ['Front-End-Https', 'X-Forwarded-Proto', /*'forwarded'*/],
```

File: tests/framework/web/RequestTest.php [hunk 2]
```diff
@@ -726,8 +759,23 @@ public function testGetUserIP($server, $expected)
                 'forwarded',
             ],
         ]);
+        $this->assertEquals($expected, $request->getUserIP());
 
+        $request = new Request([
+            'trustedHosts' => [
+                '192.168.0.0/24' => ['X-Forwarded-For', 'forwarded'],
```
Comment (kamarton) [inline tests/framework/web/RequestTest.php]: add test with 
```php
'192.168.0.0/24' => ['X-Forwarded-For', /*'forwarded'*/],
```

**Comment by samdark** [comment]
@kamarton have I added these alright in https://github.com/yiisoft/yii2/pull/17774/commits/0bb3a568d95c63c61bb0d1ae3a04d662401dc8a0? If so, build, indeed, fails.

**Comment by kamarton** [comment]
> @kamarton have I added these alright in [0bb3a56](https://github.com/yiisoft/yii2/commit/0bb3a568d95c63c61bb0d1ae3a04d662401dc8a0)? If so, build, indeed, fails.

yes, but I'm not sure with all inputs correct tests. I'll try to look at it today.

**Comment by mikk150** [comment]
those cases were literally what I was worried about

**Comment by samdark** [comment]
@shushenghong would you please check failing tests?

**Comment by shushenghong** [comment]
@samdark @mikk150 @kamarton  the failed tests are  the problem of dataProvider, such as

https://github.com/yiisoft/yii2/blob/0bb3a568d95c63c61bb0d1ae3a04d662401dc8a0/tests/framework/web/RequestTest.php#L528-L532

it should be false if `forwarded` header is not trusted

https://github.com/yiisoft/yii2/blob/0bb3a568d95c63c61bb0d1ae3a04d662401dc8a0/tests/framework/web/RequestTest.php#L601-L614

**Comment by samdark** [comment]
Right. Could you adjust it? Can do it as well but *significantly* later :(

**Comment by samdark** [comment]
Adjusted tests. Waiting for the build...

**Comment by samdark** [comment]
Confirmed that w/o `forwarded` it works as expected. Merged. Thank you.

**Comment by shushenghong** [comment]
sorry for my late reply, since i'm in china spring festival, i will add unit test, focus on testing trust headers in another pull request. 
Happy Chinese New Year, @samdark @mikk150 @kamarton

**Comment by samdark** [comment]
Happy Chinese New Year, @shushenghong!

**Pull request merged**